lib/assertions.js

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) 2021-2022 Digital Bazaar, Inc. All rights reserved.
+ */
+export function assertHttpsUrl(url) {
+  const _url = assertUrl(url);
+  assertHttps(_url);
+}
+
+export function assertUrl(url) {
+  if(url instanceof URL) {
+    return url;
+  }
+  if(typeof url === 'string') {
+    return new URL(url);
+  }
+  throw new TypeError('"url" must be a string or a url');
+}
+
+export function assertHttps(url) {
+  if(url.protocol === 'https:') {
+    return true;
+  }
+  throw new TypeError(
+    `"url" protocol must by "https:" received ${url.protocol}`);
+}
+
+export function assertDidWeb(did) {
+  const didType = typeof did;
+  if(didType !== 'string') {
+    throw new TypeError(`Expected did to be a string received ${didType}`);
```

Comment:
Can probably cap DID in these since from higher level calling code it could be confusing to see local assert param names flagged as errors.
```suggestion
    throw new TypeError(`Expected DID to be a string received ${didType}`);
```